scripts/add_new_map.js

```diff
@@ -0,0 +1,191 @@
+var _ = require('underscore'),
+  path = require('path'),
+  fs = require('fs'),
+  jsio = require('jsio'),
+  new_maps_loc = 'new_maps/',
+  bridge_height = 3,
+  coming_soon_height = 5,
+  width = 10,
+  length = 20,
+  len = 0,
+  repeat = process.argv[3] || 1,
+  chapter = process.argv[4] && parseInt(process.argv[4]) === 1 ? true : false,
+  max_ms = parseInt(process.argv[2]),
+  single_map_len = length * repeat,
+  chapter_count = max_ms / length,
+  final_map = path.join(__dirname, 'maps/final_map.js'),
+  new_config = path.join(__dirname, 'maps/final_config.json'),
+  writable = fs.createWriteStream(final_map),
+  writable_config = fs.createWriteStream(new_config),
+  new_maps = fs.readdirSync(new_maps_loc),
+  new_height = (new_maps.length * single_map_len) + (new_maps.length * bridge_height),
+  old_config = fs.readFileSync(path.join(__dirname, 'maps/tile_config.json'), 'utf8'),
+  new_map_config = [],
+  actual_height, map_no, new_item, map_data
+  get_object = function (tile, pos) {
+    var ms_obj = {
+      node: '1',
+      'friends': {
+        'position': 'bottom'
+      },
+      'tags':{
+        'milestone': true
+      }
+    };
+    ms_obj.map = pos;
+    ms_obj.id = max_ms;
+    max_ms--;
+    if (tile.x || tile.x === 0) {
+      ms_obj.x = tile.x;
+    }
+    if (tile.y || tile.y === 0) {
+      ms_obj.y = tile.y;
+    }
+    return ms_obj;
+  },
+  create_map = function () {
+    var final_map = [],
+      i = 0;
+
+    if (!_.isNumber(max_ms) && max_ms > 0) {
+      console.log("Error: Please provide total no of milestones!!");
+      return;
+    }
+
+    console.log("Starting with max ms: ", max_ms, "\nRepeat: ", repeat, "\nchapter: ", chapter);
+
+    old_config = JSON.parse(old_config.replace(/'/g,'"')
+      .replace(/(['"])?([a-z0-9A-Z_]+)(['"])?:/g, '"$2": '));
+
+    jsio.path.add('../');
+    jsio('import scripts.maps.original_map as original_map');
```

Comment:
Can we shift this statement to top

Reply:
we need to import jsio first after that only we can import. moving to top of the function.